src/components/Chart/ScatterPlots.js

```diff
@@ -0,0 +1,400 @@
+    if (props.xTicks) {
+      xValues = xValues.concat(props.xTicks)
+    }
+    const x = scales[props.xScale]()
+      .domain(extent(xValues))
+      .range([paddingLeft, paddingLeft + innerWidth])
+    const xNice = props.xNice === undefined
+      ? Math.min(Math.max(Math.round(innerWidth / 150), 3), 5)
+      : props.xNice
+    if (xNice) {
+      x.nice(xNice)
+    }
+    const xAxis = calculateAxis(props.xNumberFormat || props.numberFormat, t, x.domain()) // xUnit is rendered separately
+    const xTicks = props.xTicks || (props.xScale === 'log' ? get3EqSpaTicks(x) : xAxis.ticks)
+    // ensure highest value is last: the last value is labled with the unit
+    xTicks.sort(ascending)
+
+    // setup y axis
+    let yValues = data.map(d => d.y)
+    if (props.yTicks) {
+      yValues = yValues.concat(props.yTicks)
+    }
+    const y = scales[props.yScale]()
+      .domain(extent(yValues))
+      .range([innerHeight + paddingTop, paddingTop])
+    const yNice = props.yNice === undefined
+      ? Math.min(Math.max(Math.round(innerHeight / 150), 3), 5)
+      : props.yNice
+    if (yNice) {
+      y.nice(yNice)
+    }
+    const yAxis = calculateAxis(props.yNumberFormat || props.numberFormat, t, y.domain(), tLabel(props.yUnit))
+    const yTicks = props.yTicks || (props.yScale === 'log' ? get3EqSpaTicks(y) : yAxis.ticks)
+    // ensure highest value is last: the last value is labled with the unit
+    yTicks.sort(ascending)
+
+    const colorAccessor = d => d.datum[props.color]
+    const colorValues = data.map(colorAccessor).filter(deduplicate).filter(Boolean)
+    runSort(props.colorSort, colorValues)
+
+    const size = scaleSqrt().domain(extent(data, d => d.size)).range(props.sizeRange)
+
+    let colorRange = props.colorRanges[props.colorRange] || props.colorRange
+    if (!colorRange) {
+      colorRange = colorValues.length > 3
+        ? props.colorRanges.discrete
+        : props.colorRanges.sequential3
+    }
+    const color = scaleOrdinal(colorRange).domain(colorValues)
+
+    this.symbols = data.map((value, i) => {
+      return {
+        key: `symbol${i}`,
+        value,
+        cx: x(value.x),
+        cy: y(value.y),
+        r: size(value.size)
+      }
+    })
+
+    return (
+      <div style={{ position: 'relative' }}>
+        <svg width={width} height={height} ref={this.setContainerRef}>
+          <desc>{description}</desc>
+          {this.symbols.map((symbol, i) => (
+            <circle key={symbol.key}
+              style={{ opacity }}
+              fill={color(colorAccessor(symbol.value))}
+              cx={symbol.cx}
+              cy={symbol.cy}
+              r={symbol.r} />
+          ))}
+          {this.state.hover.map((symbol, i) => (
+            <circle key={`hover${symbol.key}`}
+              fill='none'
+              stroke='#000'
+              cx={symbol.cx}
+              cy={symbol.cy}
+              r={symbol.r} />
+          ))}
+          {
+            yTicks.map((tick, i) => (
```

Comment:
Maybe the rendering of the axes could be encapsulated in separate methods – along with the axis/ticks calculations above. Haven't tried it, though.